tensorflow/lite/experimental/shlo/ops/dot_general.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2024 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_LITE_EXPERIMENTAL_SHLO_OPS_DOT_GENERAL_H_
+#define TENSORFLOW_LITE_EXPERIMENTAL_SHLO_OPS_DOT_GENERAL_H_
+
+#include "absl/status/status.h"
+#include "tensorflow/lite/experimental/shlo/tensor.h"
+
+namespace shlo_ref {
+
+class DotGeneralOp {
+ public:
+  struct Attributes {
+    Tensor lhs_batching_dimensions;
+    Tensor rhs_batching_dimensions;
+    Tensor lhs_contracting_dimensions;
+    Tensor rhs_contracting_dimensions;
+  };
+  Attributes attributes;
+  std::vector<size_t> lhs_result_dims;
+  std::vector<size_t> rhs_result_dims;
+  std::vector<size_t> lhs_index;
+  std::vector<size_t> rhs_index;
+  std::vector<size_t> lhs_index_helper;
+  std::vector<size_t> rhs_index_helper;
+  std::vector<size_t> output_index;
+  std::vector<size_t> output_shape;
```

Comment:
We try to avoid memory allocations.

Reply:
changed to absl::InlinedVector<size_t, 6>